### PR TITLE
feat: fix post policy escape bug, update conformance tests

### PR DESCRIPTION
**Pull request by JesseLovelace**
V4 Post Policy had an issue where quotes in the json policy would get double escaped (`"` would became `\\"` instead of `\"`), this fixes that issue and updates the conformance tests to the latest version

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java [hunk 1]
```diff
@@ -443,13 +443,21 @@ public String toJson() {
...
+      char[] jsonArray = json.toCharArray();
+      for (int i = 0; i < jsonArray.length; i++) {
+        char c = jsonArray[i];
         if (c >= 128) { // is a unicode character
           escapedJson.append(String.format("\\u%04x", (int) c));
         } else {
           switch (c) {
             case '\\':
-              escapedJson.append("\\\\");
+              // The JsonObject/JsonArray operations above handle quote escapes, so leave any "/""
```
Comment (elharo) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java:454]: is forward slash meant here? Looks like it should be a backslash

File: google-cloud-storage/src/test/java/com/google/cloud/storage/V4PostPolicyTest.java [hunk 2]
```diff
@@ -166,17 +154,25 @@ public void test() {
             style);
 
     String expectedPolicy = testData.getPolicyOutput().getExpectedDecodedPolicy();
+
     StringBuilder escapedPolicy = new StringBuilder();
 
     // Java automatically unescapes the unicode escapes in the conformance tests, so we need to
     // manually re-escape them
-    for (char c : expectedPolicy.toCharArray()) {
+    char[] expectedPolicyArray = expectedPolicy.toCharArray();
```
Comment (elharo) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/V4PostPolicyTest.java:162]: This appears to be reproducing the algorithm in the test. That risks copying bugs from one place to another, and verifying the buggy behavior. Can you make the expected output a literal?

File: pom.xml [hunk 3]
```diff
@@ -93,7 +93,7 @@
       <dependency>
         <groupId>com.google.cloud</groupId>
         <artifactId>google-cloud-conformance-tests</artifactId>
-        <version>0.0.11</version>
+        <version>0.1.1</version>
```
Comment (elharo) [inline pom.xml:96]: Does this verify the change?

**Review by elharo — COMMENTED** [review]
Is there a bug reference for this?

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java (hunk 1 above)
Comment (JesseLovelace) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java:454]: Should be, will fix

File: google-cloud-storage/src/test/java/com/google/cloud/storage/V4PostPolicyTest.java (hunk 2 above)
Comment (JesseLovelace) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/V4PostPolicyTest.java:162]: Yeah, they both happen to need to do the same thing (escaping certain characters). Sorry, I'm unsure what you mean by "make it a literal," can you elaborate?

File: pom.xml (hunk 3 above)
Comment (JesseLovelace) [reply pom.xml:96]: Yeah, in fact the failures caused by it is what brought the bug to my attention (https://github.com/googleapis/java-storage/pull/510)

**Review by BenWhitehead — APPROVED** [review]
Feel free to merge after fixing the typo

**Pull request merged**